instagram/src/SearchBar.js

```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react';
+import './SearchBar.css';
+
+ const SearchBar = (props) =>  {
+        return (
+            <div className="Header">
+              <div className="Left">
+                <div className="Logo"><i className="fa fa-instagram"></i></div>
+                <div className="Branding">Instagram</div>
+              </div>
+              <div className="Middle">
+                <input type='text'value={props.search} onChange={props.handleSearch} />
```

Comment:
The `onChange` handler here would receive your function that updates the state that stores the input, not the function that executes the search itself. With this, you'll be searching on every change to the input.